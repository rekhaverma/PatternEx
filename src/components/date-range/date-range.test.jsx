import React from 'react';
import moment from 'moment';
import DateRange from './date-range.component';

/**
 * @status: WIP
 * @sign-off-by: Alex Andries
 * @todo: Review and add necessary tests
 */
describe('<DateRange />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      currentSingleDate: moment.utc('2018-01-01'),
      activeOption: '7days',
      startDate: moment.utc('2018-01-01'),
      endDate: moment.utc('2018-01-07'),
      boxIsOpen: false,
      openBox: jest.fn(),
      closeBox: jest.fn(),
    };
  });

  it('should match with snapshot default', () => {
    wrapper = shallow(<DateRange {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot for single type', () => {
    props.type = 'single';
    wrapper = shallow(<DateRange {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot for daily type', () => {
    props.type = 'daily';
    wrapper = shallow(<DateRange {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
