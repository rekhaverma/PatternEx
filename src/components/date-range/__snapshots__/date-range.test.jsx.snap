// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateRange /> should match with snapshot default 1`] = `
<div>
  <DateRange
    activeOption="7days"
    allowEnableDatesHours={false}
    boxIsOpen={false}
    className="dateRange"
    closeBox={[Function]}
    currentSingleDate={"2018-01-01T00:00:00.000Z"}
    dateFormat="YYYY-MM-DD"
    disableDays={false}
    enableDates={Array []}
    enableHours={Array []}
    endDate={"2018-01-07T00:00:00.000Z"}
    label="Date range"
    onDateChange={[Function]}
    onMonthChange={[Function]}
    onUpdateDailyDate={[Function]}
    openBox={[Function]}
    options={
      Array [
        Object {
          "id": "yesterday",
          "label": "Yesterday",
        },
        Object {
          "id": "7days",
          "label": "7 Days",
        },
        Object {
          "id": "30days",
          "label": "30 Days",
        },
        Object {
          "id": "thismonth",
          "label": "This Month",
        },
        Object {
          "id": "lastmonth",
          "label": "Last Month",
        },
        Object {
          "id": "customRange",
          "label": "Custom Range",
        },
      ]
    }
    showHoursList={true}
    startDate={"2018-01-01T00:00:00.000Z"}
    theme="column"
    type=""
    updateDateRange={[Function]}
  />
</div>
`;

exports[`<DateRange /> should match with snapshot for daily type 1`] = `
<div>
  <DateRange
    activeOption="7days"
    allowEnableDatesHours={false}
    boxIsOpen={false}
    className="dateRange"
    closeBox={[Function]}
    currentSingleDate={"2018-01-01T00:00:00.000Z"}
    dateFormat="YYYY-MM-DD"
    disableDays={false}
    enableDates={Array []}
    enableHours={Array []}
    endDate={"2018-01-07T00:00:00.000Z"}
    label="Date range"
    onDateChange={[Function]}
    onMonthChange={[Function]}
    onUpdateDailyDate={[Function]}
    openBox={[Function]}
    options={
      Array [
        Object {
          "id": "yesterday",
          "label": "Yesterday",
        },
        Object {
          "id": "7days",
          "label": "7 Days",
        },
        Object {
          "id": "30days",
          "label": "30 Days",
        },
        Object {
          "id": "thismonth",
          "label": "This Month",
        },
        Object {
          "id": "lastmonth",
          "label": "Last Month",
        },
        Object {
          "id": "customRange",
          "label": "Custom Range",
        },
      ]
    }
    showHoursList={true}
    startDate={"2018-01-01T00:00:00.000Z"}
    theme="column"
    type="daily"
    updateDateRange={[Function]}
  />
</div>
`;

exports[`<DateRange /> should match with snapshot for single type 1`] = `
<div>
  <DateRange
    activeOption="7days"
    allowEnableDatesHours={false}
    boxIsOpen={false}
    className="dateRange"
    closeBox={[Function]}
    currentSingleDate={"2018-01-01T00:00:00.000Z"}
    dateFormat="YYYY-MM-DD"
    disableDays={false}
    enableDates={Array []}
    enableHours={Array []}
    endDate={"2018-01-07T00:00:00.000Z"}
    label="Date range"
    onDateChange={[Function]}
    onMonthChange={[Function]}
    onUpdateDailyDate={[Function]}
    openBox={[Function]}
    options={
      Array [
        Object {
          "id": "yesterday",
          "label": "Yesterday",
        },
        Object {
          "id": "7days",
          "label": "7 Days",
        },
        Object {
          "id": "30days",
          "label": "30 Days",
        },
        Object {
          "id": "thismonth",
          "label": "This Month",
        },
        Object {
          "id": "lastmonth",
          "label": "Last Month",
        },
        Object {
          "id": "customRange",
          "label": "Custom Range",
        },
      ]
    }
    showHoursList={true}
    startDate={"2018-01-01T00:00:00.000Z"}
    theme="column"
    type="single"
    updateDateRange={[Function]}
  />
</div>
`;
